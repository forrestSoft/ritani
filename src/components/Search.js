import React from 'react'
import Autosuggest from './Autosuggest'
import './Search.scss'

// TODO manage auto suggest as a passed component and not spaghetti props
function Search(props){
	return(
		<section className="search">
			<span>Search for a github user by their <em>username</em>. Or not. Don't let me tell you what to do.</span>

			<Autosuggest 
				onChange={props.handleChange}
				onClick={props.handleClick}
				autocomplete_total={props.autocomplete_total}
				autocomplete_users={props.autocomplete_users}
				value={props.autocomplete_value}
				open={props.autocomplete_open}  />
		</section>
	)
}

export default Search