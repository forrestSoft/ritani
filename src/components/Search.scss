.search{
	// padding: 3vw;

	h2{
    margin-top: 0;
    margin-bottom: 1vw;
    padding: 3vw;
	  padding-bottom: 0;
  }
}