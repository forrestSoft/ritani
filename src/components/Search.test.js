import React from 'react'
import {shallow} from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import Search from './Search'

it('render matches snapshot', () => {
  const search = shallowToJson(shallow(<Search />))
  expect(search).toMatchSnapshot();
});

// it('calls passed handleChange on "onChange"', () => {
// 	const mock = jest.fn().mockReturnValue('default')
// 	const search = shallow(<Search handleChange={mock}/>)
// 	search.find('input').simulate('change')
// 	expect(mock).toBeCalled()
// })