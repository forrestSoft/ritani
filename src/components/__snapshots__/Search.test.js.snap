// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<Fragment>
  <span>
    Please search for a github user by their 
    <em>
      username
    </em>
  </span>
  <Autosuggest />
</Fragment>
`;
